### PR TITLE
Parachain staking functionality done

### PR DESCRIPTION
**Pull request by fgamundi**

### CODE REVIEW

File: packages/page-staking/package.json [hunk 1]
```diff
@@ -14,6 +14,7 @@
   "version": "0.110.2-5",
   "dependencies": {
     "@babel/runtime": "^7.17.8",
+    "@moonbeam-network/api-augment": "^0.0.1-beta.4",
```
Comment (ekenigs) [inline packages/page-staking/package.json]: As I said, not sure if we can use it.

File: packages/page-staking/src/ParachainStakingApp/index.tsx [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+
+function ParachainStakingApp ({ basePath, className = '' }: AppProps): React.ReactElement<AppProps> {
+  const { api } = useApi();
+  const { t } = useTranslation();
+
+  // summary info
+  const roundInfo = useCall<ParachainStakingRoundInfo>(api.query.parachainStaking.round);
+  const totalSelected = useCall<u32>(api.query.parachainStaking.totalSelected)?.toNumber();
+  const totalSelectedStaked = useCall<BN>(api.query.parachainStaking.staked, [roundInfo?.current]);
+  const inflation = (useCall<ParachainStakingInflationInflationInfo>(api.query.parachainStaking.inflationConfig));
```
Comment (ekenigs) [inline packages/page-staking/src/ParachainStakingApp/index.tsx]: Unnecessary braces.

File: packages/page-staking/src/ParachainStakingApp/index.tsx [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  const inflationPrct = inflation?.annual.ideal.toHuman();
+  const parachainBondInfo = useCall<ParachainStakingParachainBondConfig>(api.query.parachainStaking.parachainBondInfo);
+  const parachainBondInfoPrct = parachainBondInfo?.percent.toHuman();
+  const bestNumberFinalized = useBestNumber();
+  const collatorCommission = (useCall<Perbill>(api.query.parachainStaking.collatorCommission));
```
Comment (ekenigs) [inline packages/page-staking/src/ParachainStakingApp/index.tsx]: Again

**Review by goozang-records — APPROVED** [review]
Looks good. thank you!

To confirm, the functionality included is:
1. top dashboard with staking stats: 
* collators selected / total possible collators chosen
* number of waiting collators
* active / delegators
* total staked by selected candidates
* annual inflation
* % of inflation for parachain rent
* % collator commission
* current round number
* time until next round
2. extrinsic functionality for delegators:
* `delegate`
* `delegator_bond_more`
* `delegator_bond_less`
* `leave_delegators`
* `revoke_delegation`
* `execute_pending_delegation_request`
* `execute_leave_delegators`
* `cancel_pending_delegation_request`
* `cancel_leave_delegators`

The submission will require instructions to show all the extrinsic functionality via the UI (TODO).

File: packages/page-staking/src/ParachainStakingApp/index.tsx [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+    let _allDelegatorCount = 0;
+    let _activeDelegatorCount = 0;
+    const sorted: CandidateState[] = [];
+
+    if (!allCandidatesTopDelegations || !allCandidatesBottomDelegations || !selectedCandidates) {
+      return;
+    }
+
+    // unwrap output
+    allCandidates?.forEach(([storageKey, candidateInfoRaw], i) => {
```
Comment (ekenigs) [inline packages/page-staking/src/ParachainStakingApp/index.tsx]: When you have an array as input and you need to have another array as output it's always better to use `.map()`.

**Comment by fgamundi** [comment]
@4meta5 the extrinsics a user can do are:

- `delegate`
- `delegator_bond_more`
- `schedule_delegator_bond_less`
- `schedule_revoke_delegation`
- `execute_delegation_request`
- `cancel_delegation_request`

We don't  have a way to do `schedule_leave_delegators`, but we could add a button or something at the top if it's necessary.

**Comment by goozang-records** [comment]
It's not necessary, the functionality to leave delegators is already provided by schedule_revoke_delegation.

Sounds good, thanks!

**Pull request closed without merging**